src/statistics.jl

```diff
@@ -244,6 +301,43 @@ function mean!(
     return y
 end
 
+"""
+    mean(
+        M::Manifold,
+        x::AbstractVector,
+        [w::AbstractWeights,]
+        method::GeodesicInterpolationWithinRadius;
+        kwargs...,
+    )
+
+Estimate the Riemannian center of mass of `x` using
+[`GeodesicInterpolationWithinRadius`](@ref).
+
+See [`mean`](@ref mean(::Manifold, ::AbstractVector, ::AbstractVector, ::GeodesicInterpolation))
+for a description of `kwargs`.
+"""
+mean(::Manifold, ::AbstractVector, ::AbstractVector, ::GeodesicInterpolationWithinRadius)
+
+function mean!(
+    M::Manifold,
+    y,
+    x::AbstractVector,
+    w::AbstractVector,
+    method::GeodesicInterpolationWithinRadius;
+    shuffle_rng = nothing,
+    kwargs...,
+)
+    mean!(M, y, x, w, GeodesicInterpolation(); shuffle_rng = shuffle_rng, kwargs...)
+    radius = method.radius
+    injectivity_radius(M) ≤ radius && return y
+    for i in eachindex(x)
+        @inbounds if distance(M, y, x[i]) ≥ radius
```

Comment:
Cool! That'S a good solution here, keeping it flexible to not necessarily be the injectivity radius but still yielding a very readable code :)

Reply:
Yes, the idea is that if the radius provided is so large that it's (probably) beyond our ability to compute distances anyways, then we just skip the expensive check. But now I see that I should be passing `y` to `injectivity_radius` here. :)

Reply:
Ah yes, of course you can use the local injectivity radius and the global one you use now might be a little too conservative.